packages/webview_flutter/webview_flutter/lib/platform_interface.dart

```diff
@@ -2,547 +2,23 @@
-typedef WebViewPlatformCreatedCallback = void Function(
-    WebViewPlatformController? webViewPlatformController);
-
-/// Interface for a platform implementation of a WebView.
-///
-/// [WebView.platform] controls the builder that is used by [WebView].
-/// [AndroidWebViewPlatform] and [CupertinoWebViewPlatform] are the default implementations
-/// for Android and iOS respectively.
-abstract class WebViewPlatform {
-  /// Builds a new WebView.
-  ///
-  /// Returns a Widget tree that embeds the created webview.
-  ///
-  /// `creationParams` are the initial parameters used to setup the webview.
-  ///
-  /// `webViewPlatformHandler` will be used for handling callbacks that are made by the created
-  /// [WebViewPlatformController].
-  ///
-  /// `onWebViewPlatformCreated` will be invoked after the platform specific [WebViewPlatformController]
-  /// implementation is created with the [WebViewPlatformController] instance as a parameter.
-  ///
-  /// `gestureRecognizers` specifies which gestures should be consumed by the web view.
-  /// It is possible for other gesture recognizers to be competing with the web view on pointer
-  /// events, e.g if the web view is inside a [ListView] the [ListView] will want to handle
-  /// vertical drags. The web view will claim gestures that are recognized by any of the
-  /// recognizers on this list.
-  /// When `gestureRecognizers` is empty or null, the web view will only handle pointer events for gestures that
-  /// were not claimed by any other gesture recognizer.
-  ///
-  /// `webViewPlatformHandler` must not be null.
-  Widget build({
-    required BuildContext context,
-    // TODO(amirh): convert this to be the actual parameters.
-    // I'm starting without it as the PR is starting to become pretty big.
-    // I'll followup with the conversion PR.
-    required CreationParams creationParams,
-    required WebViewPlatformCallbacksHandler webViewPlatformCallbacksHandler,
-    WebViewPlatformCreatedCallback? onWebViewPlatformCreated,
-    Set<Factory<OneSequenceGestureRecognizer>>? gestureRecognizers,
-  });
-
-  /// Clears all cookies for all [WebView] instances.
-  ///
-  /// Returns true if cookies were present before clearing, else false.
-  Future<bool> clearCookies() {
-    throw UnimplementedError(
-        "WebView clearCookies is not implemented on the current platform");
-  }
-}
+/// Re-export the classes from the webview_flutter_platform_interface through
+/// the `platform_interface.dart` file so we don't accidentally break any
+/// non-endorsed existing implementations of the interface.
+export 'package:webview_flutter_platform_interface/webview_flutter_platform_interface.dart'
+    show
+        AutoMediaPlaybackPolicy,
+        CreationParams,
+        JavascriptChannel,
+        JavascriptChannelRegistry,
+        JavascriptMessage,
+        JavascriptMode,
+        JavascriptMessageHandler,
+        WebViewPlatform,
```

Comment:
For posterity: we accidentally shipped a breaking change to external implementors here without noticing. There's nothing we can reasonably do about it at this point, and AFAIK it only affected one client (now fixed), but what happened was:
- https://github.com/flutter/plugins/pull/4302/files#diff-bbee1da5da193a072dca86d16467e75802852826099a7b0db2e6c497c71a3f31R54 changed `WebViewPlatform`. In isolation that change seemed reasonable because it was a new package, so wasn't a breaking change.
- This switched from the old local `WebViewPlatform` to the new one, which has a new argument to `build`, which is breaking for any implementations of `WebViewPlatform`. The fact that it wasn't a direct copy isn't apparent here, so we forgot to consider that.

Because in this repository we previously had only the one combined package, and we were thus not eating our own dogfood in an important way with respect to the platform interface, nothing in CI alerted us to the mistake. The good news is that being fully federated gives us much more protection against making that mistake in the future (which is validation of the need for us to follow the intended policy of having all of our federated plugins be fully federated precisely so that we *are* eating our own dogfood).

So ironically the change that will save us from this bug in the future caused exactly the bug it's meant to avoid. But the bug was one-time, and the protection is persistent for all future changes, so it's definitely a net win despite that.